src/App.vue

```diff
@@ -402,18 +403,32 @@ export default {
         this.fetchBusinessInfo(this.businessId),
         this.fetchTasks(this.businessId),
         this.fetchFilings(this.businessId || this.tempRegNumber),
-        this.fetchAddresses(this.businessId),
         this.fetchParties(this.businessId)
       ])
 
-      if (!data || data.length !== 6) throw new Error('Incomplete business data')
+      if (!data || data.length !== 5) throw new Error('Incomplete business data')
 
       this.storeEntityInfo(data[0])
       await this.storeBusinessInfo(data[1])
       this.storeTasks(data[2])
       this.storeFilings(data[3])
-      this.storeAddresses(data[4])
-      this.storeParties(data[5])
+      this.storeParties(data[4])
+    },
+
+    async fetchAddressData ():Promise<void> {
+      let hasBAError = false
+      const data = await Promise.resolve(
+        this.fetchAddresses(this.businessId)
+      ).catch(error => {
+        if (error.response.status === 404 &&
+          error.response.config.url.includes('addresses') &&
+          error.response.data.message.includes('address not found')) hasBAError = true
+      })
+
+      let badAddress = { data: { businessOffice: null } }
+      if (!data && hasBAError) this.storeAddresses(badAddress) // if 404 and business address not found
+      if (!data && !hasBAError) throw new Error('Incomplete business data')
+      if (data) this.storeAddresses(data)
```

Comment:
This is complicated to review. I'd prefer to see...
```
if situation 1 {
  handle situation 1
} else if situation 2 {
  handle situation 2
} else {
  handle fallback/common situation, if applicable
}
```

Reply:
Other simplification ideas:

Since `fetchAddresses()` normally returns address data, why not simply return alternate address data in the catch block? Then you would only need to store the addresses after this block.

You could also move your `throw` into the catch block.

Reply:
Thanks, much easier to review/understand :)